frontend/exporter/src/rustc_utils.rs

```diff
@@ -389,8 +389,14 @@ pub fn dyn_self_ty<'tcx>(
             ty::Binder::dummy(ty::ExistentialPredicate::Projection(proj))
         });
 
-    let preds =
-        tcx.mk_poly_existential_predicates_from_iter([main_pred].into_iter().chain(ty_constraints));
+    // sort before calling `mk_poly_existential_predicates`
+    use crate::rustc_middle::ty::ExistentialPredicateStableCmpExt;
+    let mut preds: Vec<_> = [main_pred].into_iter().chain(ty_constraints).collect();
+    preds.sort_by(|a, b| {
+        a.skip_binder().stable_cmp(tcx, &b.skip_binder())
+    });
+
+    let preds = tcx.mk_poly_existential_predicates(&preds);
```

Comment:
Just a nitpick, maybe you can put the mutable let binding and the sorting directly in the top-level `let` binding, just for the sake of clarity?
```suggestion
    let preds = {
      // Stable sort predicates to prevent platform-specific ordering issues
      let mut preds: Vec<_> = [main_pred].into_iter().chain(ty_constraints).collect();
      preds.sort_by(|a, b| {
        use crate::rustc_middle::ty::ExistentialPredicateStableCmpExt;
        a.skip_binder().stable_cmp(tcx, &b.skip_binder())
      });
      tcx.mk_poly_existential_predicates(&preds)
    };
```